pkgs/by-name/az/az-pim-cli/package.nix

```diff
@@ -37,11 +38,13 @@ buildGoModule (finalAttrs: {
     ''
   );
 
-  doInstallCheck = true;
-  nativeInstallCheck = [ versionCheckHook ];
-  versionCheckProgramArg = "version";
-
-  passthru.updateScript = nix-update-script { };
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion {
```

Comment:
writableTmpDirAsHomeHook in nativeInstallCheckInputs would also work, but this is also good.

Reply:
Oh, yeah I suppose that makes sense. Otherwise it could accidentally rely on parts of the build environment.